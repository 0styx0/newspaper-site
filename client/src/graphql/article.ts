import gql from 'graphql-tag';

const ArticleQuery = gql`
    query ArticleQuery($issue: Int!, $url: String!) {
        articles(issue: $issue, url: $url) {
            id,
            article,
            canEdit,
            tags,
            comments {
                id
                content,
                dateCreated,
                canDelete,
                author {
                    fullName,
                    profileLink
                }
            }
        }
    }

`;

const ArticleCreate = gql`
    mutation ArticleCreate($tags: [String], $url: String!, $article: String!) {
        createArticle(tags: $tags, url: $url, article: $article) {
            issue
            url
        }
    }
`;

export {
    ArticleQuery,
    ArticleCreate
};
