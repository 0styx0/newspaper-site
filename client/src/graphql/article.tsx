import { gql } from 'react-apollo';

const ArticleQuery = gql`
    query ArticleQuery($issue: Int!, $url: String!) {
        articles(issue: $issue, url: $url) {
            id,
            article,
            canEdit,
            comments {
                id
                content,
                dateCreated,
                canEdit,
                author {
                    fullName,
                    profileLink
                }
            }
            tags {
                all
            }
        }
    }

`;

export {
    ArticleQuery
};
