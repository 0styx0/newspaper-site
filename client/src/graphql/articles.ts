import gql from 'graphql-tag';

const ArticleQuery = gql`

    query ArticleQuery($issue: ID) {
        issues(limit: 1, num: $issue) {
          max
          num
          articles {
            tags
            url
            id
            displayOrder
            dateCreated
            views
            author {
                fullName
                profileLink
            }
          }
        }
    }
`;

const ArticlePreviewIssueQuery = gql`

    query ArticleIssueQuery($issue: ID) {
        issues(limit: 1, num: $issue) {
          articles {
            url
            issue
            displayOrder
            views
            lede
            images(slide: true) {
                url
            }
          }
        }
    }
`;

const ArticlePreviewTagQuery = gql`

    query ArticleTagQuery($issue: Int, $tag: String) {
        articles(issue: $issue, tag: $tag) {
          url
          issue
          displayOrder
          views
          lede
          images(slide: true) {
              url
          }
        }
    }
`;

const ArticleUpdate = gql`
    mutation ArticleUpdate($data: [UpdateArticle], $password: String) {
        updateArticles(data: $data, password: $password) {
            id
            tags
            displayOrder
        }
    }
`;

const EditArticle = gql`
    mutation EditArticle($id: ID, $article: String) {
        editArticle(id: $id, article: $article) {
            id
            tags
            displayOrder
        }
    }
`;

const ArticleDelete = gql`

    mutation deleteArticles($ids: [ID], $password: String!) {
        deleteArticles(ids: $ids, password: $password) {
            id
        }
    }

`;

export {
    ArticleQuery,
    ArticlePreviewIssueQuery,
    ArticlePreviewTagQuery,
    ArticleUpdate,
    EditArticle,
    ArticleDelete
};