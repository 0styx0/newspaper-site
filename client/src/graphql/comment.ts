import { gql } from 'react-apollo';

const CommentCreate = gql`

    mutation CommentCreate($artId: ID!, $content: String!) {
        createComment(artId: $artId, content: $content) {
            id
        }
    }
`;

export {
    CommentCreate
};