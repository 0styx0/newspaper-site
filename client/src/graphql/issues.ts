import gql from 'graphql-tag';


const IssueQuery = gql`
    query IssueQuery {
        issues {
            num
            name
            views
            datePublished
            public
            canEdit
        }
    }
`;

const IssueUpdate = gql`
    mutation updateIssue($name: String, $public: Boolean, $password: String!) {
        updateIssue(name: $name, public: $public, password: $password) {
            name
            public
        }
    }
`;

const IssueInfoQuery = gql`
    query IssueInfo($num: ID) {
      issues(num: $num, limit: 1) {
          num
          max
          name
      }
    }
`;



export {
    IssueQuery,
    IssueInfoQuery,
    IssueUpdate
};
