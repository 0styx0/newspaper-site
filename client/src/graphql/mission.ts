import gql from 'graphql-tag';

const MissionEdit = gql`
    mutation editMission($mission: String!) {
        editMission(mission: $mission) {
            mission
        }
    }
`;

const MissionQuery = gql`
    query missionQuery {
        mission {
            mission
            canEdit
        }
    }
`;

export {
    MissionEdit,
    MissionQuery
};