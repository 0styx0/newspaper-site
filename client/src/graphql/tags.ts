import gql from 'graphql-tag';

const TagQuery = gql`

    query allTags {
        allTags
    }
`;

const TagCreate = gql`

    mutation createTag($tag: String!) {
        createTag(tag: $tag) {
            tag
        }
    }
`;

export {
    TagQuery,
    TagCreate
};
