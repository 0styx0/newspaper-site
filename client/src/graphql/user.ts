import { gql } from 'react-apollo';

const UserQuery = gql`

    query users($profileLink: String) {
        users(profileLink: $profileLink) {
            articles {
                id
                url
                dateCreated
                tags
                views
                issue
                canEdit
            }
            views
            level
            id
            fullName
            canEdit
        }
    }
`;

const PrivateUserQuery = gql`

    query users($profileLink: String) {
        users(profileLink: $profileLink) {
            notifications
            twoFactor
            email
            id
        }
    }
`;

const UserUpdate = gql`
    mutation updateProfile(
        $notifications: Boolean, $twoFactor: Boolean, $newPassword: String, $password: String!
    ) {
        updateProfile(
            notifications: $notifications, twoFactor: $twoFactor,
            newPassword: $newPassword, password: $password
        ) {
            id
        }
    }
`;

const RecoverPassword = gql`
    mutation recoverPassword($email: String!, $authCode: String!, $username: String!) {
        recoverPassword(email: $email, authCode: $authCode, username: $username) {
            message
        }
    }
`;

const UserLogin = gql`
    mutation login($username: String!, $password: String!) {
        login(username: $username, password: $password) {
            jwt
        }
    }
`;

// to be used only after user is logged in
const UserVerifyEmail = gql`
    mutation verifyEmail($authCode: String!) {
        verifyEmail(authCode: $authCode) {
            jwt
        }
    }
`;

const UserCreate = gql`
    mutation createUser(
        $username: String!,
        $email: String!,
        $level: Int,
        $firstName: String!,
        $middleName: String,
        $lastName: String!,
        $password: String!) {

            createUser(
                username: $username,
                email: $email,
                level: $level,
                firstName: $firstName,
                middleName: $middleName,
                lastName: $lastName,
                password: $password
            ) {
                id
            }
        }
`;

export {
    UserQuery,
    PrivateUserQuery,
    UserUpdate,
    UserLogin,
    RecoverPassword,
    UserCreate,
    UserVerifyEmail
};