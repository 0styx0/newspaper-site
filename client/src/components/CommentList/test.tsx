import * as React from 'react';
import * as sinon from 'sinon';
import CommentListContainer from './container';
import customCasual from '../../tests/casual.data';
import setFakeJwt from '../../tests/jwt.helper';
import localStorageMock from '../../tests/localstorage.mock';
import { renderWithGraphql } from '../../tests/snapshot.helper';
import mockGraphql, { mountWithGraphql, createMutation } from '../../tests/graphql.helper';
import snapData from './__snapshots__/comments.example';
import { CommentCreate } from '../../graphql/comment';
import { Comment } from './shared.interface';
const wait = require('waait');


describe('<CommentListContainer>', () => {

    function setup(
        graphql: any[] = [],
        artId: string = customCasual.word,
        comments: Comment[] = customCasual.comments
    ) {

        return mountWithGraphql(
            graphql,
            (
                <CommentListContainer
                  comments={comments}
                  artId={artId}
                />
            )
        );
    }

    describe('snapshots', () => {

        /**
         * Takes snapshots
         */
        async function snap() {

            await renderWithGraphql(
                mockGraphql(
                    [],
                    (
                        <CommentListContainer
                          comments={snapData}
                          artId={'random-id'}
                        />
                    )
                )
            );
        }

        it('renders correctly when can add', async () => {

            setFakeJwt({level: 3, id: customCasual.word});
            await snap();
        });

        it('renders correctly when cannot add', async () => {

            localStorageMock.removeItem('jwt');
            await snap();
        });
    });

    describe('#onAdd', () => {

        it('gets called after comment submission (with correct data)', async () => {

            setFakeJwt({level: 3, id: customCasual.word});
            const spy = sinon.spy(CommentListContainer.prototype, 'onAdd');

            const comment = customCasual.comments[0];
            const artId = customCasual.word;

            const wrapper = await setup(
                [
                    createMutation(
                        CommentCreate,
                        { artId, content: comment.content },
                        { createComment: { id: comment.id }}
                    )
                ],
                artId
            );

            wrapper.find('#reply .content').simulate('blur', { target: { innerHTML: comment.content } });
            wrapper.find('.editable-submit').simulate('click');
            await wait(0);

            expect(spy.calledOnce).toBeTruthy();
        });
    });
});
