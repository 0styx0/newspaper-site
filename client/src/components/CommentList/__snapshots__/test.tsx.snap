// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CommentListContainer> snapshots renders correctly when cannot add 1`] = `
<div
  id="comments"
/>
`;
